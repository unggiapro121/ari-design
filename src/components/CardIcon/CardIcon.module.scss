@import '../../styles/variables';
@import '../../styles/mixins';

.card {
  &IconBlock {
    position: relative;
    cursor: pointer;
    background: $color-black;
    line-height: 0px;
    width: 100px;

    &:hover {
      .cardIcon {
        opacity: 0.5;
      }

      .cardIconOverlay {
        opacity: 1;
      }
    }

    @include md {
      margin-right: 36px;
    }
  }

  &Icon {
    opacity: 1;
    width: 100%;
    max-height: 100%;
    height: auto;
    transition: 0.5s ease;

    @include md {
      width: 77px;
    }
  }

  &IconOverlay {
    transition: 0.5s ease;
    opacity: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
  }

  &__loadLazy {
    visibility: hidden;
    animation: fadein 1s forwards 1s;

    @keyframes fadein {
      0% {
        transform: scale(0, 0);
        visibility: visible;
        opacity: 0;
      }

      100% {
        transform: scale(1, 1);
        visibility: visible;
        opacity: 1;
      }
    }
  }
}
