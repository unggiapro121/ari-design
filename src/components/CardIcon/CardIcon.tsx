import clsx from 'clsx';
import React from 'react';

import VisibleIcon from './../../../assets/icons/icons-eye-visible.png';
import styles from './CardIcon.module.scss';

export interface CardIconProps {
  url: string;
  onClick?: () => void;
  className?: string;
  style?: React.CSSProperties;
  loadLazy?: boolean;
}

/**
 * Describe Your Component
 * @param children JSX children
 */
const CardIcon = ({ url, onClick, className = '', style, loadLazy }: CardIconProps) => {

  return (
    <div className={clsx(styles.cardIconBlock, className && className, loadLazy && styles.card__loadLazy)} style={style} data-testid={'card-Icon'} onClick={()=> onClick?.()}>
      <img src={url} alt="modal icon" className={styles.cardIcon} width={'100'} height={'100'} loading="lazy" />
      <div className={styles.cardIconOverlay}>
        <img src={VisibleIcon} alt="hover icon" width={'24'} height={'24'} loading="lazy" />
      </div>
    </div>
  );
};

export default CardIcon;