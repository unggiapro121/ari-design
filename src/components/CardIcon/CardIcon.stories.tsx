import * as React from 'react';

import CardIcon from './CardIcon';

export default {
  title: 'UI/CardIcon',
  component: CardIcon,
};

export const Primary = (args) => <CardIcon {...args} />;
Primary.args = {
  url: './../../../assets/placeholder/hero-birdseye.webp',
};