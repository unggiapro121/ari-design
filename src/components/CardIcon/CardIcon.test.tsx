import React from 'react';
import renderer from 'react-test-renderer';

import CardIcon from './CardIcon';

describe('CardIcon component', () => {
  test('does render correctly', () => {
    const component = renderer.create(<CardIcon url={'./../../../assets/placeholder/hero-birdseye.webp'}/>).toJSON();

    expect(component).toMatchSnapshot();
  });

});
