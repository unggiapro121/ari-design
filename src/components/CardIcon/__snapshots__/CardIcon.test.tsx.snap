// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardIcon component > does render correctly 1`] = `
<div
  className="_cardIconBlock_1365fc"
  data-testid="card-Icon"
  onClick={[Function]}
>
  <img
    alt="modal icon"
    className="_cardIcon_1365fc"
    height="100"
    loading="lazy"
    src="./../../../assets/placeholder/hero-birdseye.webp"
    width="100"
  />
  <div
    className="_cardIconOverlay_1365fc"
  >
    <img
      alt="hover icon"
      height="24"
      loading="lazy"
      src="/assets/icons/icons-eye-visible.png"
      width="24"
    />
  </div>
</div>
`;
